src/content/docs/guides/usage/data/parse-nested-data.mdx

```diff
@@ -173,69 +200,49 @@ content.params.forwarded_for = content.params.forwarded_for.split(", ").map(x, x
 }
 ```
 
-Let's now reshape the data, keeping only the fields we're interested in:
+## Reconstruct values into the desired form
+
+We've now reached an important state: all data is deconstructed into atomic
+values with proper types. We can't take the data apart any further. This is the
+inflection point where we shift from disassembly of the original data into
+reassembly into the desired form. For example, you could [map your events to
+OCSF](../../../../tutorials/map-data-to-ocsf).
+
+In our toy example, we'll just create a slightly flatter structure:
 
 ```tql
-load_file "sample.txt"
-read_syslog
+from {line: "2023-06-09T05:51:36.433Z mrs www[929]: I103 1703182696 request baddr=194.1.1.1 bport=58630 c_ctype=image/png caddr=194.1.1.1 cin=860 cout=898 cport=58630 duration=0.00 forwarded_for='192.1.1.1, 191.1.1.1' laddr=127.128.11.145 lport=1 method=GET paddr=172.0.0.1 pport=80 proto=6 relay_name=wwwserver response_code=200 rnum=6489895 s_ctype=image/png saddr=172.0.0.1 sin=898 sout=819 sport=80 status=OK url=http://test.de/images/icons/144x144.png"}
+this = line.parse_syslog()
 content = content.parse_grok("%{DATA:id} %{DATA:ts} %{DATA:action} %{GREEDYDATA:params}")
 content.params = content.params.parse_kv()
-// Interpret the timestamp as a seconds since Unix epoch.
 content.ts = from_epoch(content.ts * 1s)
-// Interpret the duration parameter as seconds.
 content.params.duration = content.params.duration * 1s
-// Split the forwarded_for field into an array of IP addresses.
 content.params.forwarded_for = content.params.forwarded_for.split(", ").map(x, x.ip())
 this = {
   // We keep both timestamps, labelling the one from the message `time` and the
   // one from Syslog itself `forwarded_time`:
   time: content.ts,
   forwarded_time: timestamp,
-  // Copy the other id and the action from the Syslog message:
+  host: hostname,
   id: content.id,
   action: content.action,
-  // Lastly, take all the params:
-  ...content.params,
+  source: content.params.saddr,
+  destination: content.params.saddr,
```

Comment:
Do you think it'd make sense to add back the unmapped fields as kv?

Reply:
For a full-blown mapping, yes. But I think it's distracting here.

We do this in our mapping tutorial, though.